src/EloquentRepository.php

```diff
@@ -635,4 +637,25 @@ private function getRelationsForeignKeyName(HasOneOrMany $relation): string
     {
         return method_exists($relation, 'getForeignKeyName') ? $relation->getForeignKeyName() : $relation->getPlainForeignKey();
     }
+
+    /**
+     * Get the first element against the base query.
+     *
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function first(): ?Model
```

Comment:
What motivated you to add these methods?

Reply:
The repository feels incomplete without a function to get the first result from the built-in query.

The only single-result function in the repository is `read($id)` which attempts to query a model by its id, regardless of what query was used to build the repository.

`first` and `firstOrFail` would have been useful when I was using the Repository to power the lookup of a store location by brand_id. Instead I had to modify the repository query to limit it to only one result, like this:
```
$repository->modify()->add(fn (Builder $query) => $query->whereBrandId($brand_id)->limit(1));
$repository->all()->first();
```